flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/SqlGatewayServiceImpl.java

```diff
@@ -311,8 +340,29 @@ public GatewayInfo getGatewayInfo() {
         return GatewayInfo.INSTANCE;
     }
 
+    // --------------------------------------------------------------------------------------------
+
     @VisibleForTesting
     public Session getSession(SessionHandle sessionHandle) {
         return sessionManager.getSession(sessionHandle);
     }
+
+    /** Fetch all results for configuring session. */
+    private ResultSet fetchConfigureSessionResult(
```

Comment:
It seems like this function is general and could be used to fetch the results from any `OperationHandle`, right?

Should it be factor out or maybe named something more general?

Reply:
I have discussed with @fsk119 about this and the `busy wait` problem above. The result is here:

As the [FLIP-91](https://cwiki.apache.org/confluence/display/FLINK/FLIP-91%3A+Support+SQL+Gateway) designed, the `configureSession` shouldn't return result (if it runs successfully means the result is OK; otherwise exception will be thrown). To make sure the execution has finished without error, we should use `TableResult#await` to wait if necessary until the data is ready.